src/Microsoft.Health.Blob/Registration/BlobClientRegistrationExtensions.cs

```diff
@@ -5,21 +5,22 @@
 
 using System;
 using System.Linq;
+using Azure.Storage.Blobs;
 using EnsureThat;
-using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection.Extensions;
-using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
 using Microsoft.Health.Blob.Configs;
 using Microsoft.Health.Blob.Features.Storage;
-using Microsoft.Health.Extensions.DependencyInjection;
-using Microsoft.IO;
+using Microsoft.Health.Blob.Registration;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
+    /// <summary>
+    /// Extension methods for configuring a <see cref="BlobServiceClient"/> in an <see cref="IServiceCollection"/>.
+    /// </summary>
     public static class BlobClientRegistrationExtensions
     {
-        public const string BlobStoreConfigurationSectionName = "BlobStore";
-
+        [Obsolete("Please use " + nameof(AddBlobServiceClient) + " with " + nameof(BlobClientBuilderExtensions.InitializeContainer) + " instead.")]
         public static IServiceCollection AddBlobDataStore(this IServiceCollection services, Action<BlobDataStoreConfiguration> configureAction = null)
```

Comment:
I am assuming there is no health check in azure functions.

So you need to do everything that is here and skip BlobClientProvider and ...TestProvider.
+Add(blobClientInitializer.CreateBlobClient(blobDataStoreConfiguration));
+Add(blobClientInitializer.InitializeDataStoreAsync(_blobServiceClient, blobDataStoreConfiguration, collectionInitializers));